pkg/prom/cluster/node.go

```diff
@@ -214,8 +206,8 @@ func (n *node) performClusterReshard(ctx context.Context, joining bool) error {
 			return nil, nil
 		}
 
-		ctx = user.InjectOrgID(ctx, "fake")
-		return nil, n.notifyReshard(ctx, id)
+		notifyCtx := user.InjectOrgID(c, "fake")
```

Comment:
The original code seemed like a bug, since the ctx != c context passed into this function.